File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        // String str = s.toLowerCase().replaceAll("[^a-zA-Z0-9]","");
        // String rev = new StringBuilder(str).reverse().toString();
     
        // return str.equals(rev);
        String st = s.toLowerCase().replaceAll("[^a-zA-Z0-9]","");
        return isPal(st,0,st.length()-1);
    }
    public boolean isPal(String s, int l, int r){
        if(l>=r) return true;
        if(s.charAt(l)!=s.charAt(r)){
            return false;
        }
        return isPal(s,l+1,r-1);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isAlphnum(self, s: str) -> bool:
        x = ord(s)
        if 97<=x<=122 or 48<=x<=57:
            return True
        else:
            return False
    def isPalindrome(self, s: str) -> bool:
        stt = s.lower()
        i = 0
        j = len(stt)-1
        while i<=j:
            if not self.isAlphnum(stt[i]):
                i+=1
            elif not self.isAlphnum(stt[j]):
                j-=1
            
            elif stt[i]==stt[j]:
                i+=1
                j-=1
            else:
                return False
        return True